nextcord/application_command.py

```diff
@@ -745,12 +745,50 @@ def from_callback(
                 skip_counter = 1
                 # TODO: use typing.get_type_hints when 3.9 is standard
                 typehints = typing_extensions.get_type_hints(self.callback, include_extras=True)
+                callback_params = signature(self.callback).parameters
+                possible_options: Set[str] = set()
+
+                for i, param in enumerate(callback_params.values()):
+                    if not (
+                        param.annotation is param.empty  # could be a self or interaction parameter
+                        or issubclass(
+                            param.annotation, Interaction
+                        )  # will always be an interaction parameter
+                        # TODO: use typing.Self when 3.11 is standard
+                        or param.annotation
+                        is typing_extensions.Self  # will always be a self parameter
+                        or isinstance(
+                            param.annotation, typing.TypeVar
+                        )  # will always be a self parameter
+                    ) and i in (0, 1):
+                        possible_options.add(param.name)
```

Comment:
It seems an error came about through the addition of a `not` in [`6fb2131`](https://github.com/nextcord/nextcord/pull/736/commits/6fb213165f981cfc8524f21cda3e1ad07f9db0b1).

The negation was added only on the *first* part of the condition which would make it so only the first two parameters could be added as possible options.

Here are **two** suggestions that could fix this issue. **Note:** the second suggestion is a simplified version and should be preferred over the first.

Here is what it would look like with the `not` added to outside *both* conditions instead of the first, however this can be simplified (see below)

```py
# Only add params that are not self or Interaction
if not (
    (
        # could be a self or interaction parameter
        param.annotation is param.empty
        # will always be an interaction parameter
        or issubclass(param.annotation, Interaction)
        # will always be a self parameter
        # TODO: use typing.Self when 3.11 is standard
        or param.annotation is typing_extensions.Self
        # will always be a self parameter
        or isinstance(param.annotation, typing.TypeVar)
    )
    and i in (0, 1)
):
    possible_options.add(param.name)
```

Using de-morgan simplification, we can distribute the negation:

```py
# Only add params that are not self or Interaction
if (
    # could be a self or interaction parameter
    param.annotation is not param.empty
    # will always be an interaction parameter
    and not issubclass(param.annotation, Interaction)
    # will always be a self parameter
    # TODO: use typing.Self when 3.11 is standard
    and param.annotation is not typing_extensions.Self
    # will always be a self parameter
    and not isinstance(param.annotation, typing.TypeVar)
) or i not in (0, 1):
    possible_options.add(param.name)
```

```suggestion
                    # Only add params that are not self or Interaction
                    if (
                        # could be a self or interaction parameter
                        param.annotation is not param.empty
                        # will always be an interaction parameter
                        and not issubclass(param.annotation, Interaction)
                        # will always be a self parameter
                        # TODO: use typing.Self when 3.11 is standard
                        and param.annotation is not typing_extensions.Self
                        # will always be a self parameter
                        and not isinstance(param.annotation, typing.TypeVar)
                    ) or i not in (0, 1):
                        possible_options.add(param.name)
```

Reply:
I don't really like the "distributing the negative" solution, so I'll go with the first one

Reply:
Nevermind